packages/react-router/tests/useMatch.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { afterEach, describe, expect, it, test, vi } from 'vitest'
+import '@testing-library/jest-dom/vitest'
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import {
+  Link,
+  Outlet,
+  RouterProvider,
+  createMemoryHistory,
+  createRootRoute,
+  createRoute,
+  createRouter,
+  useMatch,
+} from '../src'
+import type { RouteComponent, RouterHistory } from '../src'
+
+describe('useMatch', () => {
```

Comment:
We also probably have a type test in a `useMatch.test-d.tsx`, to check for the return type when the throw flag is set to `false` vs what it is when `true`.

Reply:
Yes, for sure we need the type test aswell